notebooks.go

```diff
@@ -363,6 +368,85 @@ func (s *server) GetNotebooks(w http.ResponseWriter, r *http.Request) {
 	s.respond(w, r, resp)
 }
 
+func (s *server) GetDefaultNotebook(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+
+	log.Printf("loading notebooks for %q", namespace)
+
+	notebooks, err := s.listers.notebooks.Notebooks(namespace).List(labels.Everything())
+	if err != nil {
+		s.error(w, r, err)
+		return
+	}
+
+	resp := &notebookapiresponse{
+		APIResponseBase: APIResponseBase{
+			Success: false,
+			Status:  http.StatusOK,
+		},
+	}
+
+	for _, notebook := range notebooks {
+
+		if val, ok := notebook.Labels["notebook.statcan.gc.ca/default-notebook"]; ok {
+			if val == "true" {
+				// Load events
+				allevents, err := s.listers.events.Events(notebook.Namespace).List(labels.Everything())
```

Comment:
I'm not a fan of the large code duplication that is happening between this new function and the GetNotebooks function https://github.com/StatCan/jupyter-apis/blob/ede3c5adb70a427437676493dfa4bd7deccdc7f1/notebooks.go#L295.

What you should do is transfer all that code that is in the for loops for both those functions to put in in its own function "processNotebook(notebook) that returns a notebookresponse object. Then, inside those for loops, you would just need to call that function, get the notebookresponse, and put that in the resp.Notebook.

I'm just thinking that if we ever need to change how we process notebooks, it will be a lot easier to do if the code is all in one place.